bundle/phases/destroy.go

```diff
@@ -26,8 +31,63 @@ func assertRootPathExists(ctx context.Context, b *bundle.Bundle) (bool, error) {
 	return true, err
 }
 
+func approvalForDestroy(ctx context.Context, b *bundle.Bundle) (bool, error) {
+	tf := b.Terraform
+	if tf == nil {
+		return false, fmt.Errorf("terraform not initialized")
+	}
+
+	// read plan file
+	plan, err := tf.ShowPlanFile(ctx, b.Plan.Path)
+	if err != nil {
+		return false, err
+	}
+
+	deleteActions := make([]terraformlib.Action, 0)
+	for _, rc := range plan.ResourceChanges {
+		if rc.Change.Actions.Delete() {
+			deleteActions = append(deleteActions, terraformlib.Action{
+				Action:       "delete",
+				ResourceType: rc.Type,
+				ResourceName: rc.Name,
+			})
+		}
+	}
+
+	if len(deleteActions) > 0 {
+		cmdio.LogString(ctx, "The following resources will be deleted:")
+		for _, a := range deleteActions {
+			cmdio.Log(ctx, a)
+		}
+		cmdio.LogString(ctx, "")
+
+	}
+
+	cmdio.LogString(ctx, fmt.Sprintf("All files and directories at the following location will be deleted: %s", b.Config.Workspace.RootPath))
+	cmdio.LogString(ctx, "")
+
+	if b.AutoApprove {
+		return true, nil
+	}
+
+	approved, err := cmdio.AskYesOrNo(ctx, "Would you like to proceed?")
+	if err != nil {
+		return false, err
+	}
+
+	return approved, nil
+}
+
 // The destroy phase deletes artifacts and resources.
 func Destroy() bundle.Mutator {
+	// Core destructive mutators for destroy. These require informed user consent.
+	destroyCore := bundle.Seq(
+		terraform.Destroy(),
+		terraform.StatePush(),
```

Comment:
(not blocking because it's the same as it was) Why do we push an empty state prior to deleting it?

Reply:
It seems related to keeping the state consistent before deleting files, to be robust against errors, before we delete `root_path`.

We used to release the lock before calling `files.Delete()`. I think it's okay to remove this, will send a followup PR.

ref:
https://github.com/databricks/cli/pull/300/files#diff-26a7ca1d9267b02564c3de57565eea8310ba2099c4df658fb902af54a36662cd

https://github.com/databricks/cli/pull/300#discussion_r1158199489